psdisk/psdisk.c

```diff
@@ -387,199 +338,188 @@ static int psdisk_updateImg(void)
 
 static int psdisk_parseToRm(void)
 {
+	struct plist_node_t *node;
 	int i;
-	struct plist_node_t *pNode;
 
 	/* Parse partition name */
-	for (i = 0; i < strlen(optarg); ++i) {
+	for (i = 0; i < strlen(optarg); i++) {
 		if (!isalnum(optarg[i])) {
-			fprintf(stderr,"Invalid partition name - %s.\n", optarg);
+			fprintf(stderr, "Invalid partition name - %s.\n", optarg);
 			return -1;
 		}
 	}
 
-	if ((pNode = (struct plist_node_t *)calloc(1, sizeof(struct plist_node_t))) == NULL) {
-		fprintf(stderr,"Cannot allocate memory, err: %s.\n", strerror(errno));
+	node = calloc(1, sizeof(struct plist_node_t));
+	if (node == NULL) {
+		fprintf(stderr, "Cannot allocate memory, err: %s.\n", strerror(errno));
 		return -1;
 	}
 
-	if (strlen(optarg) >= sizeof(pNode->pHeader.name)) {
-		fprintf(stderr,"Invalid partition name - %s.\n", optarg);
-		free(pNode);
+	if (strlen(optarg) >= sizeof(node->part.name)) {
+		fprintf(stderr, "Invalid partition name - %s.\n", optarg);
+		free(node);
 		return -1;
 	}
 	else {
-		strncpy((char *)pNode->pHeader.name, optarg, sizeof(pNode->pHeader.name));
+		strncpy((char *)node->part.name, optarg, sizeof(node->part.name));
 	}
```

Comment:
I prpose something like this:
```c

	/* Parse partition name */
	for (i = 0; optarg[i] != '\0'; i++) {
		if (!isalnum(optarg[i]) || (i >= sizeof(node->part.name))) {
			fprintf(stderr, "Invalid partition name - %s.\n", optarg);
			return -1;
		}
	}

	node = calloc(1, sizeof(struct plist_node_t));
	if (node == NULL) {
		fprintf(stderr, "Cannot allocate memory, err: %s.\n", strerror(errno));
		return -1;
	}

	strcpy((char *)node->part.name, optarg);
```
Instead of using the global variable `optarg`, I suggest to define the function as:
```c
int psdisk_parseToRm(char *arg)
{
}
```
and use the local variable `arg`.
This applies to other functions as well.

Reply:
I agree with the proposed changes. Applied in the next force-push.